File: library/src/main/java/com/github/sundeepk/compactcalendarview/CollapsingAnimation.java
package com.github.sundeepk.compactcalendarview;


import android.view.animation.Animation;
import android.view.animation.Transformation;

class CollapsingAnimation extends Animation {
    private final int targetHeight;
    private final CompactCalendarView view;
    private int targetGrowRadius;
    private final boolean down;
    private CompactCalendarController compactCalendarController;

    public CollapsingAnimation(CompactCalendarView view, CompactCalendarController compactCalendarController, int targetHeight, int targetGrowRadius, boolean down) {
        this.view = view;
        this.compactCalendarController = compactCalendarController;
        this.targetHeight = targetHeight;
        this.targetGrowRadius = targetGrowRadius;
        this.down = down;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        float grow = 0;
        int newHeight;
        if (down) {
            newHeight = (int) (targetHeight * interpolatedTime);
            grow = (interpolatedTime * (targetGrowRadius * 2));
        } else {
            float progress = 1 - interpolatedTime;
            newHeight = (int) (targetHeight * progress);
            grow = (progress * (targetGrowRadius * 2));
        }
        compactCalendarController.setGrowProgress(grow);
        view.getLayoutParams().height = newHeight;
        view.requestLayout();

    }

    @Override
    public void initialize(int width, int height, int parentWidth,
                           int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }
}

File: library/src/main/java/com/github/sundeepk/compactcalendarview/CollapsingAnimation.kt
package com.github.sundeepk.compactcalendarview

import android.view.animation.Animation
import android.view.animation.Transformation

internal class CollapsingAnimation(private val view: CompactCalendarView, private val compactCalendarController: CompactCalendarController, private val targetHeight: Int, private val targetGrowRadius: Int, private val down: Boolean) : Animation() {
    override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
        val grow: Float
        val newHeight: Int
        if (down) {
            newHeight = (targetHeight * interpolatedTime).toInt()
            grow = interpolatedTime * (targetGrowRadius * 2)
        } else {
            val progress = 1 - interpolatedTime
            newHeight = (targetHeight * progress).toInt()
            grow = progress * (targetGrowRadius * 2)
        }
        compactCalendarController.setGrowProgress(grow)
        view.layoutParams.height = newHeight
        view.requestLayout()
    }

    override fun willChangeBounds(): Boolean {
        return true
    }
}
